vector/src/main/java/im/vector/app/features/permalink/PermalinkHandler.kt

```diff
@@ -82,22 +110,56 @@ class PermalinkHandler @Inject constructor(private val activeSessionHolder: Acti
     private fun PermalinkData.RoomLink.getRoomId(): Single<Optional<String>> {
         val session = activeSessionHolder.getSafeActiveSession()
         return if (isRoomAlias && session != null) {
-            // At the moment we are not fetching on the server as we don't handle not join room
-            session.rx().getRoomIdByAlias(roomIdOrAlias, false).subscribeOn(Schedulers.io())
+            session.rx().getRoomIdByAlias(roomIdOrAlias, true).subscribeOn(Schedulers.io())
         } else {
             Single.just(Optional.from(roomIdOrAlias))
         }
     }
 
+    private fun PermalinkData.RoomLink.getRoomAliasOrNull(): String? {
+        return if (isRoomAlias) {
+            roomIdOrAlias
+        } else {
+            null
+        }
+    }
+
     /**
      * Open room either joined, or not
      */
-    private fun openRoom(context: Context, roomId: String?, eventId: String?, buildTask: Boolean) {
+    private fun openRoom(
+            context: Context,
+            roomId: String?,
+            permalinkData: PermalinkData.RoomLink,
+            buildTask: Boolean
+    ) {
         val session = activeSessionHolder.getSafeActiveSession() ?: return
-        return if (roomId != null && session.getRoom(roomId) != null) {
-            navigator.openRoom(context, roomId, eventId, buildTask)
-        } else {
-            navigator.openNotJoinedRoom(context, roomId, eventId, buildTask)
+        if (roomId == null) {
+            context.toast(R.string.room_error_not_found)
+            return
+        }
+        val roomSummary = session.getRoomSummary(roomId)
+        val eventId = permalinkData.eventId
+        val roomAlias = permalinkData.getRoomAliasOrNull()
+        return when {
+            roomSummary?.membership?.isActive().orFalse() -> {
+                navigator.openRoom(context, roomId, eventId, buildTask)
+            }
+            roomSummary?.membership != Membership.BAN     -> {
+                val roomPreviewData = RoomPreviewData(
+                        roomId = roomId,
+                        eventId = eventId,
+                        roomAlias = roomAlias ?: roomSummary?.canonicalAlias,
+                        roomName = roomSummary?.displayName,
+                        avatarUrl = roomSummary?.avatarUrl,
+                        buildTask = buildTask,
+                        homeServers = permalinkData.viaParameters
+                )
+                navigator.openRoomPreview(context, roomPreviewData)
+            }
+            else                                          -> {
+                context.toast(R.string.error_opening_banned_room)
```

Comment:
I would have deal with this error first, and then, in the else block handle the nominal case.